datafusion/physical-plan/src/async_func.rs

```diff
@@ -221,6 +234,49 @@ impl ExecutionPlan for AsyncFuncExec {
     }
 }
 
+struct CoalesceInputStream {
+    input_stream: Pin<Box<dyn RecordBatchStream + Send>>,
+    batch_coalescer: LimitedBatchCoalescer,
+}
+
+impl Stream for CoalesceInputStream {
+    type Item = Result<RecordBatch>;
+
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<Option<Self::Item>> {
+        let mut completed = false;
+
```

Comment:
Do we want to track `elapsed_compute` time for coalescing? e.g. https://github.com/apache/datafusion/blob/887aa9f8c641cfcac4053da3dd5ad3c4b7ffe425/datafusion/physical-plan/src/coalesce_batches.rs#L296

Reply:
I had omitted it as we were not doing it already, but if it makes sense then sure! Pushing a commit for it.